common/math-transforms.js

```diff
@@ -22,16 +22,20 @@
            Since these are in tree-order,
            if we process them
            in reverse order we should side-step
-           the gnarliest of potentialnesting
+           the gnarliest of potential nesting
            issues, I think
         */
           matches.reverse().forEach(el => {
-            el.setAttribute("data-math-transformed", "");
-            let copy = el.cloneNode(true);
-            let transformed = transformer(copy);
-  
-            recurse = true;
-            el.replaceWith(transformed || copy);
+            const nextChild = el.nextSibling;
+            const parent = el.parentElement;
+            let transformed = transformer(el);
+            if (transformed && transformed !== el) {
+              transformed.setAttribute("data-math-transformed", "");
+              if (el.parentElement === parent) {
+                parent.removeChild(el);
+              }
+              parent.insertBefore(transformed, nextChild);
```

Comment:
Maybe we can review this file together? I have to admit I'm not sure what these changes are doing - I don't think this is ready for merge.  Is there a point to depth or recurse anymore?  The data-math-transformed is in pursuit of tracking that too, also unnecessary? do we just not deal with the idea that transforms could place in something which also needs transforms?  Is there a way parent could not === parent element?  etc..  I think we can wrap these questions up pretty quick, but it would probably make sense to sit together and discuss (and probably document I guess)